elooplib/eventloop.hpp

```diff
@@ -20,14 +20,20 @@ class EventLoop
     int max_pollfd_pos;
     std::queue<int> available_pos;
     std::vector<pollfd> fds;
-    std::list<std::pair<size_t, std::shared_ptr<Node>>> nodes; //TODO no pair, just remove from vector
-    //ThreadPool th_pool; TODO
+    std::list<std::pair<size_t, std::shared_ptr<Node>>> nodes; // TODO no pair, just remove from vector
+    std::list<std::pair<size_t, std::shared_ptr<Node>>> wait_q;
+    std::shared_ptr<ThreadPool> th_pool;
+
 public:
     EventLoop(size_t);
     void addNode(std::shared_ptr<Node> &, bool);
-    using nodeRetType = decltype(std::begin(nodes));
-    nodeRetType rmNode(nodeRetType &);
+    using nodeType = decltype(*std::begin(nodes));
```

Comment:
use `decltype(nodes)::value_type` and so on:
https://en.cppreference.com/w/cpp/container/vector